WalletWasabi/CoinJoin/Coordinator/Coordinator.cs

```diff
@@ -221,23 +221,46 @@ public async Task MakeSureTwoRunningRoundsAsync(Money? feePerInputs = null, Mone
 
 				int confirmationTarget = await AdjustConfirmationTargetAsync(lockCoinJoins: true).ConfigureAwait(false);
 
+				var configuredTimeout = TimeSpan.FromSeconds(RoundConfig.InputRegistrationTimeout);
 				if (runningRoundCount == 0)
 				{
-					var round = new CoordinatorRound(RpcClient, UtxoReferee, RoundConfig, confirmationTarget, RoundConfig.ConfirmationTarget, RoundConfig.ConfirmationTargetReductionRate);
+					var round = new CoordinatorRound(RpcClient, UtxoReferee, RoundConfig, confirmationTarget, RoundConfig.ConfirmationTarget, RoundConfig.ConfirmationTargetReductionRate, configuredTimeout);
 					round.CoinJoinBroadcasted += Round_CoinJoinBroadcasted;
 					round.StatusChanged += Round_StatusChangedAsync;
 					await round.ExecuteNextPhaseAsync(RoundPhase.InputRegistration, feePerInputs, feePerOutputs).ConfigureAwait(false);
 					Rounds.Add(round);
 
-					var round2 = new CoordinatorRound(RpcClient, UtxoReferee, RoundConfig, confirmationTarget, RoundConfig.ConfirmationTarget, RoundConfig.ConfirmationTargetReductionRate);
+					var round2 = new CoordinatorRound(RpcClient, UtxoReferee, RoundConfig, confirmationTarget, RoundConfig.ConfirmationTarget, RoundConfig.ConfirmationTargetReductionRate, 2 * configuredTimeout);
 					round2.StatusChanged += Round_StatusChangedAsync;
 					round2.CoinJoinBroadcasted += Round_CoinJoinBroadcasted;
 					await round2.ExecuteNextPhaseAsync(RoundPhase.InputRegistration, feePerInputs, feePerOutputs).ConfigureAwait(false);
 					Rounds.Add(round2);
 				}
 				else if (runningRoundCount == 1)
 				{
-					var round = new CoordinatorRound(RpcClient, UtxoReferee, RoundConfig, confirmationTarget, RoundConfig.ConfirmationTarget, RoundConfig.ConfirmationTargetReductionRate);
+					// If only one round is running, get the remaining time
+					var remainingTimeout = Rounds.FirstOrDefault(x => x.Status == CoordinatorRoundStatus.Running)?.RemainingInputRegistrationTime ?? TimeSpan.Zero;
+
+					TimeSpan? timeout = null;
+					if (remainingTimeout <= configuredTimeout)
+					{
+						// If it's smaller or equal than the configured timeout,
+						// then the new round's timeout should happen remaining + configured timeout.
+						timeout = remainingTimeout + configuredTimeout;
+					}
+					else if (remainingTimeout > 2 * configuredTimeout)
+					{
+						// If the remaining timeout is larger then 2 times the configured timeout,
+						// then configuration has changed, so make the configured timeout the new round's timeout.
+						timeout = configuredTimeout;
+					}
+					else
+					{
+						// Else halven the remaining timeout.
+						timeout = remainingTimeout / 2;
```

Comment:
In the last two cases, you are creating a round before the ongoing round, correct? If yes: where will new input registrations go? Does the client prioritize where to register according to the remaining time? If the answer is no then we have a problem with this.

1. Imagine that there is a round1 with 1.5h remaining 50ppl. 
2. round2 create with 45min.
3. round1 eating all the new input reg but still not reach 100 ppl. 
4. round2 will be empty and triggered after 45 min.

Reply:
We are just getting the first round naively, I guess - "thinking that, that round will be the next" 

https://github.com/molnard/WalletWasabi/blob/c218614fd0ea1fe4644c919c273e2bb3b8df734d/WalletWasabi/CoinJoin/Client/Rounds/ClientState.cs#L312

Reply:
The sync response from the backend not ordering the Rounds either. https://github.com/molnard/WalletWasabi/blob/c218614fd0ea1fe4644c919c273e2bb3b8df734d/WalletWasabi.Backend/Controllers/ChaumianCoinJoinController.cs#L67

Reply:
I knew this is a problem, but I didn't find a simple solution. Your ordering idea is great and non-intrusive thanks! Done.